src/kernel/chainparams.cpp

```diff
@@ -84,7 +84,8 @@ class CMainParams : public CChainParams {
         consensus.BIP66Height = 363725; // 00000000000000000379eaa19dce8c9b722d46ae6a57c2f1a988119488b50931
         consensus.CSVHeight = 419328; // 000000000000000004a1b34462cb8aeebd5799177f7a29cf28f2d1961716b5b5
         consensus.SegwitHeight = 481824; // 0000000000000000001c8018d9cb3b742ef25114f27563e3fc4a1902167f9893
-        consensus.MinBIP9WarningHeight = 483840; // segwit activation height + miner confirmation window
+        // Preferably to be updated with every release to avoid checking older blocks for warnings
```

Comment:
IMO, it makes sense for it to to be updated to the day of the release with the assumption that the main chain does not contain any warning bits up until then. What are your thoughts?

Reply:
Setting it to `N` will prevent warnings that would otherwise be triggered by version bits being set in blocks up to `N+2015`; so you pick a value where a reorg back to `N+2015` or lower is sufficiently unlikely (for similar values of "sufficiently unlikely" as we'd use to bury successfully activated soft forks) and that `getblockchaininfo` from above that height doesn't already include an `Unknown new rules activated` warning. (If it does, it's obviously important to verify that the new rules that were signalled aren't something we should enforce first. Once that's figured out, it's then fine to ignore the signalling).

Reply:
(Maybe it would be better to do `std::max(MinBIP9WarningHeight, nMinerConfirmationWindow) - nMinerConfirmationWindow` in `WarningBitsConditionChecker::BeginTime()` ; then you're just ignoring version bits prior to `MinBIP9WarningHeight`)

Reply:
I think `std::max(MinBIP9WarningHeight, nMinerConfirmationWindow)` is unnecessary since I doubt that `nMinerConfirmationWindow` would be higher than `MinBIP9WarningHeight`. As for the rest, it makes sense since it removes the burden of having to keep in mind that a parameter would need adjusting in the future. I initially changed it to a timestamp due to your comment on the other PR :smile:.

Reply:
It's more that `MinBIP9WarningHeight` might be lower, 0 in particular.